app/lib/server/handler.ts

```diff
@@ -0,0 +1,107 @@
+import { JobRequest, JobStorage } from '@/api'
+import { Context as RunnerContext } from './runner'
+import { Context as ServerContext } from './server'
+import * as Runner from './runner'
+
+export interface Context extends RunnerContext, ServerContext {
+  jobs: JobStorage
+}
+
+export const create = async (ctx: Context) => {
+  return new Handler(ctx)
+}
+
+class Handler {
+  storacha
+  telegram
+  cipher
+  #jobs
+  #queueFn
+
+  constructor ({ storacha, telegram, cipher, jobs, queueFn }: Context) {
+    this.storacha = storacha
+    this.telegram = telegram
+    this.cipher = cipher
+    this.#jobs = jobs
+    this.#queueFn = queueFn
+  }
+
+  async queueJob(request: JobRequest) {
+    const job = await this.#jobs.find(request.jobID)
+    if (!job) throw new Error(`job not found: ${request.jobID}`)
+
+    await this.#jobs.replace({
+      id: job.id,
+      status: 'queued',
+      params: job.params,
+      created: job.created
+    })
+    this.#queueFn(request)
+  }
+
+  async handleJob(request: JobRequest) {
+    const job = await this.#jobs.find(request.jobID)
+    if (!job) throw new Error(`job not found: ${request.jobID}`)
+    // check if job was cancelled
+    if (job.status != 'queued') {
+      return
```

Comment:
Throw? This is unexpected state.

Reply:
not really, the job can be cancelled until it's in progress